### PR TITLE
fix(button): 562 button responsive long text

### PR DESCRIPTION
**Pull request by sKudum**
## Related Issue
Closes SAP/fundamental-styles#562

## Description
long text on the button will be truncated and shown in ellipse formate based on the responsive mode.

### Before:

Screen Shot 2020-01-22 at 10 21 26 PM

### After:
Screen Shot 2020-01-22 at 10 22 51 PM

### CODE REVIEW

**Comment by stefanoScalzo** [comment]
Why is it that for popover it still overflows?

**Comment by sKudum** [comment]
> Why is it that for popover it still overflows?

Could you please elaborate -Popover.
Because this fix is to truncate only long text present inside a button.

**Comment by sKudum** [comment]
Hi @InnaAtanasova ,

I did not test the animation on the button for this change in the playground. 
Because below is my understanding.
1). A component that has an animation effect that has to be tested with an Animation option.
2).Themes: in case I did any color changes then I need to test it with different themes.
3).Touch: needs to tested when my component has a touch feature as well. 
Other options like normalize, unnormalized, RTL, compact are tested.

File: docs/pages/components/button.md [hunk 1]
```diff
@@ -240,3 +240,13 @@ Button with multiple actions
 <button class="fd-button--compact fd-button--menu">Compact</button>
 {% endcapture %}
 {% include display-component.html component=button-menu %}
+
+## Button With Text Overflow
+
+Content overflow in the button will be shown with ellipsis
+
+{% capture button %}
+<button class="fd-button">Default button with longer text</button>
```
Comment (droshev) [inline docs/pages/components/button.md]: I don't see ellipsis in this example
Screen Shot 2020-01-23 at 8 33 10 AM

Comment (sKudum) [reply docs/pages/components/button.md]: Hi Deno, 
     As requested in the bug truncation is applied based on the parent container width.
For example.
When we decrease screen width in inspect mode mobile view or tab with or with lesser screen width we can see the truncation effect.

Comment (sKudum) [reply docs/pages/components/button.md]: Sure, In button.md I will remove the duplicate example and update with a different examples.

Comment (droshev) [reply docs/pages/components/button.md]: > Hi Deno,
> As requested in the bug truncation is applied based on the parent container width.
> For example.
> When we decrease screen width in inspect mode mobile view or tab with or with lesser screen width we can see the truncation effect.

What is the value of adding another example if we cannot show it in the documentation.
Btw the menu button spec has more requirements in terms of `max-width`

Comment (sKudum) [reply docs/pages/components/button.md]: Added example with lesser parent width which shows the truncation.

Comment (sKudum) [reply docs/pages/components/button.md]: As per firori when we have a menu in the split button then the max-width is 12rem after that truncation has to be applied. For this, we have a bug.
https://github.com/SAP/fundamental-styles/issues/602.

**Comment by droshev** [comment]
@sKudum can you wait for the PR https://github.com/SAP/fundamental-styles/pull/509 to be merged as it is refactoring the button. Also the playground tests don't bring much value

**Comment by sKudum** [comment]
> @sKudum can you wait for the PR #509 to be merged as it is refactoring the button. Also the playground tests don't bring much value

Sure, will wait for #509 and regarding playground test cases- we don't need to have them at least for few button options

**Comment by sKudum** [comment]
> Does it make sense to double the visual tests for that? also i don't see ellipsis anywhere

Changes done

**Comment by droshev** [comment]
> > Does it make sense to double the visual tests for that? also i don't see ellipsis anywhere
> 
> Changes done

I still see 120 files for the visual tests

**Comment by droshev** [comment]
@sKudum I think it is better to close the PR until we get #509 merged

**Comment by sKudum** [comment]
> @sKudum I think it is better to close the PR until we get #509 merged

Hi @droshev , May I know the reason please.

**Comment by droshev** [comment]
> > @sKudum I think it is better to close the PR until we get #509 merged
> 
> Hi @droshev , May I know the reason please.

The `button` component is under refactoring and most probably what you want to achieve will done with  #509. 
Also, the fix in this PR is not covered by any test and it doesn't work for the `menu button`

**Comment by sKudum** [comment]
Closing this PR as suggested by Deno.

**Pull request closed without merging**